Utility/Utilities.cs

```diff
@@ -87,18 +94,20 @@ public partial class Utilities {
             var response = await HttpRequestAsync(host, port, "/status", HttpMethod.Get, string.Empty, ct);
             var status = await response.Content.ReadAsStringAsync(ct);
 
-            var dic = StatusRegex().Matches(status).Cast<Match>().ToDictionary(x => x.Groups[1].Value.Trim(), x => x.Groups[2].Value.Trim());
+            var dic = StatusRegex()
+                .Matches(status)
+                .Cast<Match>()
+                .ToDictionary(x =>
+                    x.Groups[1].Value.Trim(), x => x.Groups[2].Value.Trim()
+                );
+
             Logger.Trace(string.Join(Environment.NewLine, dic.Select(pair => $"{pair.Key} => {pair.Value}")));
 
             return dic;
         }
 
-        public static bool Pwi4CheckMountConnected(string host, ushort port, CancellationToken ct) {
-            var status = new Dictionary<string, string>();
-
-            Task.Run(async () => {
-                status = await Pwi4GetStatus(host, port, ct);
-            }, ct).Wait(ct);
+        public static async Task<bool> Pwi4CheckMountConnected(string host, ushort port, CancellationToken ct) {
+            var status = await Task.Run(() => Pwi4GetStatus(host, port, ct));
```

Comment:
.Wait was unnecessary. just await the task and mark the method as async.

Reply:
If you don't need it to run a separate task here you can also remove the Task.Run. I just kept this as i'm not sure in which context the caller thread runs